src/aspire/image/image.py

```diff
@@ -593,15 +593,18 @@ def filter(self, filter):
         #
         # Second note, filter dtype may not match image dtype.
         filter_values = xp.asarray(
-            filter.evaluate_grid(self.resolution), dtype=self.dtype
+            filter.evaluate_grid(self.resolution), dtype=np.float64
         )
 
         # Convolve
-        im_f = fft.centered_fft2(xp.asarray(im._data))
+        _im = xp.asarray(im._data, dtype=np.float64)
+        im_f = fft.fftshift(fft.fft2(_im), axes=(-1, -2))
         im_f = filter_values * im_f
-        im = fft.centered_ifft2(im_f)
+        im = fft.ifft2(fft.ifftshift(im_f, axes=(-1, -2)))
```

Comment:
Do we need to worry about the interplay of this with images that have been downsampled using non-legacy downsample, ie. that uses the centered_fft and maintains a different centering convention?

Reply:
I hope not!  There are clearly no failing tests for such a thing....

I'm not sure what interplay you mean. Does `downsample` use this `filter` code somehow that I've missed? Did you check this code in isolation and find a problem like it is changing the image center?  It would seem to me if _this_ filter was `ones` _this_ convolve is identity up to FFT error for both evens and odds.  ASPIRE's centered_fft wrappers are questionable if you're asking me...

Do you mean the downsample filter scaling?  That should work (part of why had to begrudgingly leave the `omega` stuff). LMK.

If for some reason we need to keep the centered_fft, I think we will not be able to repro ...

Reply:
Ok, the downsample filter scaling appears to be scaling the frequency grid correctly. I think I was just misunderstanding some notion of interplay here. Sorry about that.